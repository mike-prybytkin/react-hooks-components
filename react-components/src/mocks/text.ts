const mockText = {
  homeLink: 'home',
  aboutLink: 'about',
  cardTitle: 'Fjallraven',
  searchPlaceholder: 'Search...',
  linkBackToHome: 'Back to home page',
  heading404: 'Error 404!',
  headingAboutUs: 'About Us!',
  headingMain: 'Welcome to us!',
  pageNotReady: 'Sorry, page not ready yet...',
  loading: 'Loading ...',
  itemNotFound: 'Sorry, item not found...',
  headingForm: 'Create User!',
  notCreatedUserInForm: 'Sorry, no users have been created yet...',
  labelUserName: 'Name:',
  placeholderUserName: 'Your Name',
  labelUserBirthday: 'Birthday:',
  labelUserSalary: 'Your monthly income:',
  switcherGender: 'Gender:',
  switcherGenderMale: 'Male',
  switcherGenderFemale: 'Female',
  labelUserMailing: 'Receive mailing',
  labelFileUpload: 'Add your photo',
  createUserButton: 'Create user',
  productCardButton: 'Learn more',
  prevButton: 'prev',
  nextButton: 'next',
  userSex: 'sex:',
  userBDay: 'b-day:',
  userSalary: 'salary:',
  subscribed: 'Subscribed',
  unsubscribed: 'Unsubscribed',
  toMailings: 'to mailings!',
  fromMailings: 'from mailings...',
};

export default mockText;
