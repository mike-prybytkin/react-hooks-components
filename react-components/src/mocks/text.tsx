const mockText = {
  homeLink: 'home',
  aboutLink: 'about',
  cardTitle: 'Fjallraven',
  searchPlaceholder: 'Search...',
  linkBackToHome: 'Back to home page',
  heading404: 'Error 404!',
  headingAboutUs: 'About Us!',
  headingMain: 'Welcome to our store!',
  pageNotReady: 'Sorry, page not ready yet...',
  loading: 'Loading ...',
  itemNotFound: 'Sorry, item not found...',
  headingForm: 'Create User!',
  notCreatedUserInForm: 'Sorry, no users have been created yet...',
  labelUserName: 'Name:',
  placeholderUserName: 'Your Name',
  labelUserBirthday: 'Birthday:',
  labelUserSalary: 'Your monthly income:',
  switcherGender: 'Gender:',
  switcherGenderMale: 'Male',
  switcherGenderFemale: 'Female',
  labelUserMailing: 'Receive mailing',
  labelFileUpload: 'Add your photo',
  createUserButton: 'Create user',
};

export default mockText;
