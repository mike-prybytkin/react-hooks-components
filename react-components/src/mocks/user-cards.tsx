import { IUserCard } from 'share/types';
import { userPhoto } from './user-photo';

export const mockUserCards: IUserCard[] = [
  {
    name: 'Max',
    avatarPath: userPhoto.max,
    gender: 'Male',
    birthday: '12-08-1995',
    salary: 'less than 100$',
    mailing: true,
  },
  {
    name: 'Kate',
    avatarPath: userPhoto.kate,
    gender: 'Female',
    birthday: '01-12-2003',
    salary: '200$ - 299$',
    mailing: false,
  },
];
