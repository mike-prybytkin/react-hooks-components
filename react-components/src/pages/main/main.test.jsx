import React from 'react';
import { render, screen } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import Main from './main';
import { mockCards } from '../../mocks/cards';
import mockText from '../../mocks/text';

const setUp = () =>
  render(<Main cards={mockCards} heading={mockText.headingMain} />, { wrapper: BrowserRouter });

describe('Main component', () => {
  it('should correctly render heading', () => {
    setUp();
    const mainHeading = screen.getByText(new RegExp(mockText.headingMain, 'i'));
    expect(mainHeading).toBeInTheDocument();
  });

  it('should create snapshot', () => {
    const { asFragment } = setUp();
    expect(asFragment()).toMatchSnapshot();
  });
});
