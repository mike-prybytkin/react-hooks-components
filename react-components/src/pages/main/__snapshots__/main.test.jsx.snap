// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main component should create snapshot 1`] = `
<DocumentFragment>
  <h1>
    Welcome to our store!
  </h1>
  <div
    class="cards-wrapper"
  >
    <div
      class="card"
    >
      <h3
        class="card__title"
      >
        Fjallraven - Foldsack No. 1 Backpack, Fits 15 Laptops
      </h3>
      <img
        alt="men's clothing"
        class="card__img"
        src="https://fakestoreapi.com/img/81fPKd-2AYL._AC_SL1500_.jpg"
      />
      <p
        class="card__rating"
      >
        3.9 ★
      </p>
      <p
        class="card__price"
      >
        109.95 $
      </p>
    </div>
    <div
      class="card"
    >
      <h3
        class="card__title"
      >
        Mens Casual Premium Slim Fit T-Shirts 
      </h3>
      <img
        alt="men's clothing"
        class="card__img"
        src="https://fakestoreapi.com/img/71-3HjGNDUL._AC_SY879._SX._UX._SY._UY_.jpg"
      />
      <p
        class="card__rating"
      >
        4.1 ★
      </p>
      <p
        class="card__price"
      >
        22.3 $
      </p>
    </div>
  </div>
</DocumentFragment>
`;
